pkg/kv/kvserver/allocator/storepool/store_pool_test.go

```diff
@@ -1029,3 +1030,84 @@ func TestNodeLivenessLivenessStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestStorePoolStoreHealthTracker(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	ctx := context.Background()
+	st := cluster.MakeTestingClusterSettings()
+	stopper, _, clock, sp, _ := CreateTestStorePool(ctx, st,
+		TestTimeUntilStoreDead, false, /* deterministic */
+		func() int { return 1 }, /* nodeCount */
+		livenesspb.NodeLivenessStatus_LIVE)
+	defer stopper.Stop(ctx)
+	desc := roachpb.StoreDescriptor{
+		StoreID: 1,
+		Node:    roachpb.NodeDescriptor{NodeID: 1},
+		Capacity: roachpb.StoreCapacity{
+			IOThreshold: admissionpb.IOThreshold{
+				L0NumSubLevels:          5,
+				L0NumSubLevelsThreshold: 20,
+				L0NumFiles:              5,
+				L0NumFilesThreshold:     1000,
+			},
+		},
+	}
+
+	checkIOOverloadScore := func() float64 {
+		detail, ok := sp.GetStoreDescriptor(1)
+		require.True(t, ok)
+		score, _ := detail.Capacity.IOThreshold.Score()
+		return score
+	}
+
+	// Intiailly sublevels is 5/20 = 0.25, expect that score.
+	sp.storeDescriptorUpdate(desc)
+	require.Equal(t, 0.25, checkIOOverloadScore())
+	// Now increase the numfiles to be greater (500/1000) so the score should
+	// update to 0.5.
+	desc.Capacity.IOThreshold.L0NumFiles = 500
+	sp.storeDescriptorUpdate(desc)
+	require.Equal(t, 0.5, checkIOOverloadScore())
+	// Despite the numfiles now decreasing again to 0, the max tracked should
+	// be still 500 and therefore the score unchanged.
+	desc.Capacity.IOThreshold.L0NumFiles = 0
+	sp.storeDescriptorUpdate(desc)
+	require.Equal(t, 0.5, checkIOOverloadScore())
+	// Update the threshold, which will be used with the max tracked score
+	// (500/500).
+	desc.Capacity.IOThreshold.L0NumFilesThreshold = 500
+	sp.storeDescriptorUpdate(desc)
+	require.Equal(t, 1.0, checkIOOverloadScore())
+	// Update the sublevels to be greater than the num files score.
+	desc.Capacity.IOThreshold.L0NumSubLevels = 40
```

Comment:
Some strategic line breaks will make this test less daunting to read. Also, I don't think this test is really doing us any favours -- it's a thinly veiled test for the max sliding window aggregators you have underneath. I'd be ok nuking it.

Reply:
Removed this test.